lib/core/src/Cardano/Wallet/Primitive/Types/TokenPolicy.hs

```diff
@@ -157,3 +171,57 @@ newtype AssetLogo = AssetLogo
     deriving (Show) via (Quiet AssetLogo)
 
 instance NFData AssetLogo
+
+-- | The validated URL for the asset.
+newtype AssetURL = AssetURL
+    { unAssetURL :: URI
+    } deriving (Eq, Ord, Generic)
+    deriving (Show) via (Quiet AssetURL)
+
+instance NFData AssetURL
+
+validateMinLength :: Int -> Text -> Either String Text
+validateMinLength n text
+    | len >= n = Right text
+    | otherwise = Left $ "Length must be at least " ++ show n ++ " characters, got " ++ show len
+  where
+    len = T.length text
+
+validateMaxLength :: Int -> Text -> Either String Text
+validateMaxLength n text
+    | len <= n = Right text
+    | otherwise = Left $ "Length must be no more than " ++ show n ++ " characters, got " ++ show len
+  where
+    len = T.length text
+
+validateMetadataName :: Text -> Either String Text
+validateMetadataName = validateMinLength 1 >=> validateMaxLength 50
+
+validateMetadataAcronym :: Text -> Either String Text
+validateMetadataAcronym = validateMinLength 2 >=> validateMaxLength 4
+
+validateMetadataDescription :: Text -> Either String Text
+validateMetadataDescription = validateMaxLength 500
+
+validateMetadataURL :: Text -> Either String AssetURL
+validateMetadataURL = fmap AssetURL .
+    (validateMaxLength 250 >=> validateURI >=> validateHttps)
+  where
+      validateURI = maybe (Left "Not an absolute URI") Right
+          . parseAbsoluteURI
+          . T.unpack
+      validateHttps u@(uriScheme -> scheme)
+          | scheme == "https:" = Right u
+          | otherwise = Left $ "Scheme must be https: but got " ++ scheme
+
+validateMetadataUnit :: AssetUnit -> Either String AssetUnit
+validateMetadataUnit assetUnit@AssetUnit{name} =
+    (validateMinLength 1 name >>= validateMaxLength 30) $> assetUnit
+
+validateMetadataLogo :: AssetLogo -> Either String AssetLogo
+validateMetadataLogo logo
+    | len <= maxLen = Right logo
+    | otherwise = Left $ "Length must be no more than " ++ show maxLen ++ " bytes, got " ++ show len
+  where
+    len = BS.length $ unAssetLogo logo
+    maxLen = 87400
```

Comment:
`maxLen` should be 65535 (64KB) here right? 87400 is the max-length of the base64-encoded data payload.

Reply:
OK